File: js/recursive_selection_sort.js
function recursiveSelectionSort(array, l, r) {
    if (l < r) {
        let currentMin = array[l];
        let currentMinIndex = l;

        for (let i = l + 1; i < array.length; i++) {
            if (currentMin > array[i]) {
                currentMin = array[i];
                currentMinIndex = i;
            }
        }
        array[currentMinIndex] = array[l];
        array[l] = currentMin;
        recursiveSelectionSort(array, l + 1, r);
    }
}

function testRecursiveSelectionSort() {
    output.innerHTML += "<h3>testing recursive selection sort in javascript...</h3>";
    const unsortedArray = getRandArray(10, 1);
    output.innerHTML += `<p>unsorted array -> [${unsortedArray}]</p>`;
    recursiveSelectionSort(unsortedArray, 0, unsortedArray.length);
    output.innerHTML += `<p>sorted array -> [${unsortedArray}]</p>`;
}

testRecursiveSelectionSort();

File: python/recursive_selection_sort.py
def recursive_selection_sort(array, l, r):
    if l < r:
        currentMin = array[l]
        currentMinIndex = l
        for i in range(l + 1, len(array)):
            if currentMin > array[i]:
                currentMin = array[i]
                currentMinIndex = i
        
        array[currentMinIndex] = array[l]
        array[l] = currentMin
        recursive_selection_sort(array, l + 1, r)

print("testing recursive selection sort in python...")

unsortedArray = [4, 8, 1, 5, 2, 8, 3, 9, 7, 6]
print("unsorted array -> " + str(unsortedArray))
recursive_selection_sort(unsortedArray, 0, len(unsortedArray))
print("sorted array -> " + str(unsortedArray))